### DISCUSSION TITLE
Error installing PiHole

### DISCUSSION THREAD

**Discussion by tjcrandley** [unifi-utilities/unifi-common#182] [Q&A]
Hi all - I just upgraded my newly acquired UDMP to Firmware 1.9.3, and now I'm trying these utilities (specifically PiHole).  I am following the steps but I'm getting hung up on "Step 6 Execute /mnt/data/on_boot.d/10-dns.sh".  I get an error stating: "ln: failed to create symbolic link '/etc/cni/net.d/20-dns.conflist': No such file or directory" and then a number of follow-on errors.  I checked and the /etc/cni/net.d folder doesn't exist.  Should I mkdir this folder, or should it have been created during another process?

**Comment by tjcrandley** [comment]
Another observation:  ip isn't installed on my UDMP by default, and it's needed by the 10-dns.sh script.  Maybe I missed an "install these things via apt" step?

**Reply by telnetdoogie** [reply]
did you install 05-cni-plugins as well?
https://github.com/boostchicken/udm-utilities/blob/master/cni-plugins/05-install-cni-plugins.sh
